### PR TITLE
Initialize RuntimeConfig module only if config is specified

### PR DESCRIPTION
**Pull request by annanay25**
Signed-off-by: Annanay <annanayagarwal@gmail.com>

**What this PR does**:
This PR is a follow up from #2092. Updated so that the `RuntimeConfig` module no longer initializes if the config path is empty.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- NA Tests updated
- NA Documentation added
- NA `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/cortex/modules.go [hunk 1]
```diff
@@ -122,6 +122,10 @@ func (t *Cortex) initRing() (serv services.Service, err error) {
 
 func (t *Cortex) initRuntimeConfig() (services.Service, error) {
 	if t.Cfg.RuntimeConfig.LoadPath == "" {
+		if t.Cfg.LimitsConfig.PerTenantOverrideConfig == "" {
+			// no need to initialize module if load path is empty
+			return nil, nil
```
Comment (pstibrany) [inline pkg/cortex/modules.go:127]: I'm pretty sure leaving `t.RuntimeConfig` set to nil will cause problems elsewhere.

Comment (pstibrany) [reply pkg/cortex/modules.go:127]: Nitpick, but can we move this check *after* first check?

- First `if` -- check whether to use new or old-style options
- Second `if` -- if LoadPath is still empty, return quickly

The way it is done right now mixes these two concerns, that should be separate.

File: pkg/cortex/modules.go [hunk 2]
```diff
@@ -122,6 +122,10 @@ func (t *Cortex) initRing() (serv services.Service, err error) {
...
+		}
 		t.Cfg.RuntimeConfig.LoadPath = t.Cfg.LimitsConfig.PerTenantOverrideConfig
 		t.Cfg.RuntimeConfig.ReloadPeriod = t.Cfg.LimitsConfig.PerTenantOverridePeriod
 	}
```
Comment (pstibrany) [inline pkg/cortex/modules.go:131]:
```suggestion
	if t.Cfg.RuntimeConfig.LoadPath == "" {
		t.Cfg.RuntimeConfig.LoadPath = t.Cfg.LimitsConfig.PerTenantOverrideConfig
		t.Cfg.RuntimeConfig.ReloadPeriod = t.Cfg.LimitsConfig.PerTenantOverridePeriod
	}

	if t.Cfg.RuntimeConfig.LoadPath == "" {
		// no need to initialize module if load path is empty
		return nil, nil
	}
```

File: pkg/cortex/modules.go (hunk 1 above)
Comment (annanay25) [reply pkg/cortex/modules.go:127]: Sure, thanks @pstibrany! Making the changes.

**Pull request merged**